Sources/Contour/Transport/RabbitMQ/Internal/RabbitBus.cs

```diff
@@ -499,62 +410,45 @@ private void StartTask(CancellationToken cancellationToken)
             this.ComponentTracker.StartAll();
 
             this.logger.Trace(m => m("{0}: marking as ready.", this.Endpoint));
-            IsStarted = true;
-            ready.Set();
+            this.IsStarted = true;
+            this.ready.Set();
 
             this.OnStarted();
         }
 
         protected override void Restart(bool waitForReadiness = true)
         {
-            lock (logger)
+            lock (this.logger)
             {
                 if (this.isRestarting.WaitOne(0) || this.IsShuttingDown)
                 {
                     return;
                 }
-                ready.Reset();
-                isRestarting.Set();
+                this.ready.Reset();
+                this.isRestarting.Set();
             }
 
-            this.logger.Trace(m => m("{0}: Restarting...", Endpoint));
+            this.logger.Trace(m => m("{0}: Restarting...", this.Endpoint));
 
             this.ResetRestartTask();
 
-            var token = cancellationTokenSource.Token;
-            restartTask = Task.Factory.StartNew(() => this.StopTask(token), token, TaskCreationOptions.LongRunning, TaskScheduler.Default)
+            var token = this.cancellationTokenSource.Token;
+            this.restartTask = Task.Factory.StartNew(() => this.StopTask(token), token, TaskCreationOptions.LongRunning, TaskScheduler.Default)
                 .ContinueWith(_ => this.StartTask(token), token, TaskContinuationOptions.LongRunning, TaskScheduler.Default)
                 .ContinueWith(
                     t =>
                         {
-                            isRestarting.Reset();
+                            this.isRestarting.Reset();
                             if (t.IsFaulted)
                             {
                                 throw t.Exception.InnerException;
                             }
                         });
-/*
-               .ContinueWith(
-                   t =>
-                       {
-                           if (!ComponentTracker.CheckHealth())
-                           {
-                               Task.Factory.StartNew(
-                                   () =>
-                                       {
-                                           Thread.Sleep(1000);
-                                           this.Restart(false);
-                                       });
-                           }
-                       });
-*/
 
             if (waitForReadiness)
             {
-                restartTask.Wait(5000);
+                this.restartTask.Wait(5000);
```

Comment:
There is inconsistency between waitFor**Readyness** flag, the IsStarted property or ready MRE and the state of the restartTask (which is being awaited here).